Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -894,6 +920,374 @@ void node_implementation_init(nmos::node_model& model, slog::base_gate& gate)
+        {
+            auto data = nmos::details::make_nc_worker(temperature_sensor_control_class_id, oid, true, owner, role, value::string(user_label), description, touchpoints, runtime_property_constraints, true);
+            data[temperature] = value::number(temperature_);
+            data[unit] = value::string(unit_);
+
+            return nmos::control_protocol_resource{ nmos::is12_versions::v1_0, nmos::types::nc_worker, std::move(data), true };
+        };
+
+        // example root block
+        auto root_block = nmos::make_root_block();
+
+        nmos::nc_oid oid = nmos::root_block_oid;
+
+        // example device manager
+        auto device_manager = nmos::make_device_manager(++oid, model.settings);
+
+        // example class manager
+        auto class_manager = nmos::make_class_manager(++oid, control_protocol_state);
+
+        // example stereo gain
+        const auto stereo_gain_oid = ++oid;
+        auto stereo_gain = nmos::make_block(stereo_gain_oid, nmos::root_block_oid, U("stereo-gain"), U("Stereo gain"), U("Stereo gain block"));
+
+        // example channel gain
+        const auto channel_gain_oid = ++oid;
+        auto channel_gain = nmos::make_block(channel_gain_oid, stereo_gain_oid, U("channel-gain"), U("Channel gain"), U("Channel gain block"));
+        // example left/right gains
+        auto left_gain = make_gain_control(++oid, channel_gain_oid, U("left-gain"), U("Left gain"), U("Left channel gain"));
+        auto right_gain = make_gain_control(++oid, channel_gain_oid, U("right-gain"), U("Right gain"), U("Right channel gain"));
+        // add left-gain and right-gain to channel gain
+        nmos::push_back(channel_gain, left_gain);
+        nmos::push_back(channel_gain, right_gain);
+
+        // example master-gain
+        auto master_gain = make_gain_control(++oid, channel_gain_oid, U("master-gain"), U("Master gain"), U("Master gain block"));
+        // add channel-gain and master-gain to stereo-gain
+        nmos::push_back(stereo_gain, channel_gain);
+        nmos::push_back(stereo_gain, master_gain);
+
+        // example example-control
+        auto example_control = make_example_control(++oid, nmos::root_block_oid, U("ExampleControl"), U("Example control worker"), U("Example control worker"),
+            value::null(),
+            // specify the level 2: runtime constraints, see https://specs.amwa.tv/ms-05-02/branches/v1.0.x/docs/Constraints.html
+            // use of make_nc_property_constraints_string and make_nc_property_constraints_number to create runtime constraints
+            value_of({
+                { nmos::details::make_nc_property_constraints_string({3, 2}, 5, U("^[a-z]+$")) },
+                { nmos::details::make_nc_property_constraints_number({3, 3}, 10, 100, 2) }
+            }),
+            example_enum::Undefined,
+            U("test"),
+            3,
```

Comment:
This is a bug - the numberProperty is initialized to 3, but the runtime constraints specify a step of 2.  When the "invasive" tests run it stores the original value (3) and attempts to reset to the original value at the end of the test, which fails as it violates the runtime constraints.
```suggestion
            4,
```